### ISSUE TITLE
Enhancement: modify the action of the stretch/extent button

### ISSUE THREAD

**Issue by severinmenard** [fieldpapers/fieldpapers#145]
This is actually an old request: https://github.com/stamen/fieldpapers/issues/294
When clicking on this button:
![selection_019](https://cloud.githubusercontent.com/assets/1291055/8071987/6b732e8a-0eff-11e5-80e7-e15e38782938.png)
it actually zooms to the extent covered by the FP "size panel" (is there a better wording?). It should rather do the contrary: extent the FP "size panel" to the current zoom. It would spare a lot of "zoomin-resize-click the button" loops to the user.

**Comment by mojodna** [comment]
@sconnelley what do you think about this?

**Comment by sconnelley** [comment]
I am not sure I fully understand `extent the FP "size panel" to the current zoom`?  Right now it fits the selection box into the map viewport. How would extend the selection box to the current zoom work?

**Comment by severinmenard** [comment]
I suggest that when clicking on https://cloud.githubusercontent.com/assets/1291055/8071987/6b732e8a-0eff-11e5-80e7-e15e38782938.png it adapts the FP extent to the current zoom so that the user can see immediately what would be the FP. 
The current action is that you get back to the existing FP extent, what has almost no interest at all

**Comment by sconnelley** [comment]
@severinmenard -- I understand the button your referencing, I am still unclear of what your asking. Can you explain `adapts the FP extent to the current zoom` a little more?

**Comment by severinmenard** [comment]
so far you have to reduce the size of the FP, then zoom, then reduce, zoom,
etc.
This is tedious and would be better to directly zoom to the right zoom then
clicking on the button you are talking about to get the FP fits the area
you zoomed in. Currently if you click on it, you get back where you reduce
for the last time (or the initial zoom if did not reduce the FP)

**Comment by sconnelley** [comment]
:bulb:, I gotcha. So the selection box would expand to the map viewpoint (give or take a buffer).  I like that idea, but should we provide functionality as an additional tool?

**Comment by ian-ross** [comment]
I wonder how this would interact with a fix for another issue I've been thinking about -- https://github.com/fieldpapers/fieldpapers/issues/150 is about the fact that the map view on the atlas composer page is really pretty small, and I was thinking that it might look better to have that page be more like the main OSM page, where more or less the entire browser viewport is taken up by a map view.  I guess what @severinmenard is suggesting here would still be better behaviour than the current behaviour: you'd pan and zoom the map window to see the region you wanted, click the "zoom to extent" button to fit the FP page composer frame to the viewport, then adjust the number of atlas pages (assuming we implement the functionality of https://github.com/fieldpapers/fieldpapers/issues/146 as well, which I think is a good idea).

**Comment by severinmenard** [comment]
Sorry for not having been able to describe it clearly as you just did: yes
this is why I was talking about. I personally think the current button is
not useful and actually provides an unexpected action for the user (damned,
why am I back there?! may be the common reaction :), but this is only MHO

**Comment by smarshy** [comment]
@ian-ross With respect to this issue,what did the specified button actually intend to do?Was the aim to just resize the atlas tool to dimensions that fit the viewport without altering the zoom level and affecting the url?

**Comment by ian-ross** [comment]
@smarshy At the moment that button zooms the *map* extent to fit around the page composer frame. If I understand right, @severinmenard believes this is exactly the wrong way round for it to work: the button should resize the page composer to fit the map extents, without altering the viewed area of the map. (@severinmenard - if I've misrepresented what you were thinking, please put me right!) I'm inclined to agree with @severinmenard - it seems to make more sense to zoom the map to your region of interest first before starting to manipulate the page composer.

**Comment by severinmenard** [comment]
@ianross you got my point

**Comment by EyePlusPlus** [comment]
@ian-ross I wrote a small patch  that would eliminate the "zoomin-resize-click the button" loops to the user as mentioned in this issue. It might also possibly lead to the removal of the stretch/extent button in question (not very sure about that though)

Here's the branch I've pushed it to, in case you wanna review it
[issue145-patch](https://github.com/EyePlusPlus/fp-web/commit/f97b5c7748b0ff7e5d6df4073e93c81d7bc3e706?diff=unified)

The idea is to keep the atlas extent constant irrespective of the zoom level

Eg:
**At zoom level 12, width=388, height=150**
![screenshot from 2015-10-22 00 23 21](https://cloud.githubusercontent.com/assets/7201095/10647330/3e74e81e-7855-11e5-90df-035305e49f28.png)

**At zoom level 13, width=388, height=150**
![screenshot from 2015-10-22 00 24 08](https://cloud.githubusercontent.com/assets/7201095/10647395/b46008c4-7855-11e5-9884-26340e04afe3.png)

**Comment by ian-ross** [comment]
Thanks @EyePlusPlus - I'll take a look at that.  I want to get our UI/UX person to review the interaction model of the Field Papers UI once we get started, based on the existing setup, the issues that people have raised and what the Stamen people think.  That should give us an idea of how best to deal with all these issues.

**Comment by mojodna** [comment]
@EyePlusPlus would it be possible for you to push up a static version of that page so we can play with it interactively?

**Comment by EyePlusPlus** [comment]
@mojodna  Does [this](http://eyeplusplus.github.io/fieldpapers-demo/) help?
Also would like to mention that this static version was created after merging the patches for issue #142 and issue #145
I've committed the patches as 2 different branches because my patch for issue #142 might not be entirely correct

**Comment by mojodna** [comment]
@EyePlusPlus awesome.

@severinmenard and @sconnelley should check it out to see if it matches their behavioral expectations.

From my end, I noticed that it creates a new leaflet layer each time![selection_019](https://cloud.githubusercontent.com/assets/1291055/8071987/6b732e8a-0eff-11e5-80e7-e15e38782938.png) is clicked, which adds an additional scale bar and extra tiles.

**Comment by smarshy** [comment]
@ian-ross Keeping in mind the approach taken by @EyePlusPlus to do this, I have tried to do the same .[This](https://github.com/smarshy/fp-web/commit/fcdf9dfdb425b9914c0277bf16635929f20bb97a) is a link to my implementation of keeping the atlas extent constant.
The map still zooms to the bounds on clicking the button. I am unsure if this is what the user wants.I have tried to mute out the action of the button concerned with [a small change](https://github.com/smarshy/fp-web/commit/10081f81c2e257fa12a9d904a859d59817d7c370) 

Having said that, the current issue, which was to change the functionality of the button, still hasn't been addressed. A couple of questions though. 
Does this button do what it intended to do initially? Did users find it helpful? If they did then instead of replacing it, we could add a new button and try to address the issue with that .

**Comment by ian-ross** [comment]
There are changes in the pipeline that will help with the map/grid interaction in the compose page more generally: https://github.com/fieldpapers/fp-web/pull/23

**Comment by severinmenard** [comment]
Not sure if the fp-web#23 is now online, just tested it and the button still has the same behaviour that is not really satisfying: Eg if a user wants to create a FP atlas over a few blocks over Dakar, she/he will type Dakar at the first step and get a small scale map over the city, then will have to reduce the size of the atlas to the minimum that is still much larger than the final atlas will be, click on the button we are talking about, and repeat this as often as required to get the final extent. IMHO it will be more effective to let the user zoom (side note: seems the scrolling button can not be used) to its area of interest then getting a first atlas proposition by clicking on a button. Thinking about it, it would lead to the need of dividing buttons for columns and rows, while the current buttons add new ones, so it is not an small thing to fix.

**Comment by ian-ross** [comment]
@severinmenard The PR isn't yet deployed to the main site.  Have you looked at the fieldpapers-demo.cadasta.org site, which is set up to demonstrate the new functionality?

**Comment by severinmenard** [comment]
Yes sorry I saw it only today. IMHO option 3 (or 1) is really great.

**Comment by ian-ross** [comment]
This issue should be fixed or rendered obsolete by https://github.com/fieldpapers/fp-web/pull/23: if anyone thinks it's still a problem, please reopen this issue!
